nntrainer/graph/graph_core.h

```diff
@@ -156,25 +153,45 @@ class GraphCore {
    * @brief     get begin iterator for the forwarding
    * @retval    const iterator marking the begin of forwarding
    */
-  inline const_iterator cbegin() { return Sorted.cbegin(); }
+  template <
+    typename T = GraphNode,
+    std::enable_if_t<std::is_base_of<GraphNode, T>::value, T> * = nullptr>
+  inline graph_iterator<T const> cbegin() {
+    return graph_iterator<T const>(&Sorted[0]);
+  }
 
   /**
    * @brief     get end iterator for the forwarding
    * @retval    const iterator marking the emd of forwarding
    */
-  inline const_iterator cend() { return Sorted.cend(); }
+  template <
+    typename T = GraphNode,
+    std::enable_if_t<std::is_base_of<GraphNode, T>::value, T> * = nullptr>
+  inline graph_iterator<T const> cend() {
+    return graph_iterator<T const>((&Sorted.back()) + 1);
```

Comment:
can we use Sorted.end() here?

Reply:
i think this is already done in a latter in this PR.